bids/layout/layout.py

```diff
@@ -454,36 +459,36 @@ def _get_fieldmaps(self, path):
         files = self.get(subject=sub, suffix=suffix,
                          extensions=['nii.gz', 'nii'])
         for file in files:
-            metadata = self.get_metadata(file.filename)
+            metadata = self.get_metadata(file.path)
```

Comment:
Minor optimization would be to avoid getting that getattr (which internally does if, recurses getattr, and then sure getitem; so much for not much of action) by assigning `filepath = file.path` here and using throughout. In general, now it is something to keep in mind, it might become major optimization in tight loops

Reply:
Noted, thanks!